django/thunderstore/account/authentication.py

```diff
@@ -0,0 +1,21 @@
+import datetime
+
+from rest_framework.authentication import TokenAuthentication as DRFTokenAuthentication
+from rest_framework.authtoken.models import Token
+
+from thunderstore.account.models import ServiceAccount
+
+
+class TokenAuthentication(DRFTokenAuthentication):
+    keyword = "Bearer"
+
+    def authenticate(self, request):
+        out = super().authenticate(request)
+        if out is not None and all(out):
+            # The request has been authenticated
+            token: Token = out[1]
+            service_account = ServiceAccount.objects.filter(user=token.user).first()
```

Comment:
Was thinking about `.get(user=...)` but if we use the same token authentication system for OAuth then we would have to often fall back on a try except. In the future maybe we could create a custom `Token` class that allows us to specify what the `Token` is for, a service account or a user, which could improve performance to avoid querying both. Another idea could be a custom `User` model and adding a property to mark if it's a service account or a normal user, which we could check using `token.user`.